src/lib.rs

```diff
@@ -348,24 +348,35 @@ impl OpenzwaveAdapter {
             for notification in rx {
                 //debug!("Received notification {:?}", notification);
                 match notification {
-                    ZWaveNotification::ControllerReady(controller)  => {
-                        let service = format!("OpenZWave-{:08x}", controller.get_home_id());
+                    ZWaveNotification::ControllerReady(_controller) => {}
+                    ZWaveNotification::NodeNew(_node)               => {}
+                    ZWaveNotification::NodeAdded(node)              => {
+                        let service = format!("OpenZWave-{:08x}-{:02x}", node.get_home_id(), node.get_id());
                         let service_id = TaxId::new(&service);
-                        controller_map.push(service_id.clone(), controller);
+                        node_map.push(service_id.clone(), node);
+
+                        let mut service = Service::empty(service_id.clone(), adapter_id.clone());
+                        service.properties.insert(String::from("name"), node.get_name());
+                        service.properties.insert(String::from("product_name"), node.get_product_name());
+                        service.properties.insert(String::from("manufacturer_name"), node.get_manufacturer_name());
+                        service.properties.insert(String::from("location"), node.get_location());
 
-                        box_manager.add_service(Service::empty(service_id.clone(), adapter_id.clone()));
+                        box_manager.add_service(service);
+                    }
+                    ZWaveNotification::NodeNaming(_node)             => {
+                        // unfortunately we can't change a service' properties :(
+                        // https://github.com/fxbox/taxonomy/issues/97
+                        // When it's done we can move the properties change from above to here.
                     }
-                    ZWaveNotification::NodeNew(_node)               => {}
-                    ZWaveNotification::NodeAdded(_node)             => {}
                     ZWaveNotification::NodeRemoved(_node)           => {}
                     ZWaveNotification::ValueAdded(vid)              => {
```

Comment:
I think that we need to wait longer than this. There are still a bunch of node queries which haven't been done yet.

Certainly, by the time AllAwakeNodesQueried(?sp) then we could advertise. Otherwise commands sent too early will fail.

I'd probably create an issue and leave that for a different PR.